pkg/kv/kvserver/rafttrace/rafttrace.go

```diff
@@ -492,34 +506,18 @@ func (r *RaftTracer) iterCovered(from, to kvpb.RaftIndex, visit func(*traceValue
 // only allow MsgAppResp to be logged once per peer, and only one
 // MsgStorageAppendResp.
 func (r *RaftTracer) traceIfPast(m raftpb.Message) {
-	if m.Reject {
```

Comment:
while you're here, the first sentence of the method comment seems off

> // traceIfPast will log the message to all registered traceValues the message is
// past.

I don't understand what it's trying to tell me. I don't understand what `traceIfPast`/`traceIfCovered` means either.

Reply:
(PS: just nits, this isn't related to this PR)

Reply:
Agree, this whole file could use a good refactor/rework. I'd like to do it at some point to make raft tracing more solid. Will punt it in this PR, and improve it separately.

Reply:
`traceIfPast / traceIfCovered` just reflects the fact that some raft events are "high watermarks", while some can be directly attributed to the traced entries.

Events like `MsgAppResp` are watermarks / `traceIfPast`. The first time we see an event that "passed" out index, we trace "this index is now covered" kind of thing, and [remember](https://github.com/cockroachdb/cockroach/blob/fc46a1380068d4e9dc08801dad3505c5ba8cf611/pkg/kv/kvserver/rafttrace/rafttrace.go#L65-L73) this fact so that we don't do it twice.

Events like "apply entries 100-200" can be directly intersected with all the traced entries, and so `traceIfCovered` is used for them.